src/Mod/Sketcher/Gui/DrawSketchController.h

```diff
@@ -0,0 +1,573 @@
+class DrawSketchController
+{
+public:
+    /** @name Meta-programming definitions and members */
+    //@{
+    using ControllerBase = void;  // No base controller for parent class.
+
+    using HandlerType = HandlerT;
+    using SelectModeType = SelectModeT;
+    using ContructionMethodType = ConstructionMethodT;
+    static constexpr const int AutoConstraintInitialSize = PAutoConstraintSize;
+    //@}
+
+    /** @name Convenience definitions */
+    //@{
+    using DSDefaultHandler =
+        DrawSketchDefaultHandler<HandlerT, SelectModeT, PAutoConstraintSize, ConstructionMethodT>;
+    using ConstructionMachine = ConstructionMethodMachine<ConstructionMethodT>;
+
+    using ConstructionMethod = ConstructionMethodT;
+    using SelectMode = SelectModeT;
+    //@}
+
+protected:
+    HandlerT* handler;           // real derived type
+    bool init = false;           // true if the controls have been initialised.
+    bool firstMoveInit = false;  // true if first mouse movement not yet performed (resets)
+
+    Base::Vector2d prevCursorPosition;
+    Base::Vector2d lastControlEnforcedPosition;
+
+    int onViewIndexWithFocus = 0;  // track the index of the on-view parameter having the focus
+    int nOnViewParameter = OnViewParametersT::defaultMethodSize();
+
+
+    /** @name Named indices for controlling on-view controls */
+    //@{
+    enum OnViewParameter
+    {
+        First,
+        Second,
+        Third,
+        Fourth,
+        Fifth,
+        Sixth,
+        Seventh,
+        Eighth,
+        Ninth,
+        Tenth,
+        nOnViewParameters  // Must Always be the last one
+    };
+    //@}
+
+    std::vector<std::unique_ptr<Gui::EditableDatumLabel>> onViewParameters;
+
+    /// Class to keep track of colors used by the on-view parameters
+    class ColorManager
+    {
+    public:
+        SbColor dimConstrColor, dimConstrDeactivatedColor;
+
+        ColorManager()
+        {
+            init();
+        }
+
+    private:
+        void init()
+        {
+            ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath(
+                "User parameter:BaseApp/Preferences/View");
+
+            dimConstrColor = SbColor(1.0f, 0.149f, 0.0f);
+            dimConstrDeactivatedColor = SbColor(0.8f, 0.8f, 0.8f);
+
+            float transparency = 0.f;
+            unsigned long color = (unsigned long)(dimConstrColor.getPackedValue());
+            color = hGrp->GetUnsigned("ConstrainedDimColor", color);
+            dimConstrColor.setPackedValue((uint32_t)color, transparency);
+
+            color = (unsigned long)(dimConstrDeactivatedColor.getPackedValue());
+            color = hGrp->GetUnsigned("DeactivatedConstrDimColor", color);
+            dimConstrDeactivatedColor.setPackedValue((uint32_t)color, transparency);
+        }
+    };
+
+public:
+    /** Creates the controller.
+     *  @param dshandler a controllable DSH handler
+     */
+    DrawSketchController(HandlerT* dshandler)

```

Comment:
Please add an explicit here, as in ```explicit DrawSketchController(```

This is to avoid implicit conversions and it is a very good thing.